test/src/lib/node/JournaledDAO.js

```diff
@@ -71,5 +76,32 @@ describe('JournaledDAO', function() {
     });
   });
 
+  it('should read from journaled file and populate arrayDao properly', function(done) {
+
+    foam.dao.JournaledDAO.create({
+      delegate: cars,
+      journal: foam.dao.NodeFileJournal.create({
+        fd: fs.openSync(filename, 'r+')
+      })
+    })
+    .select({
+      put: function(sub, data) {
+        cars.put(data);
```

Comment:
This should be redundant. The JournaledDAO will already copy all of the journal data into the delegate (cars) when it's created.